### ISSUE TITLE
Order data missing payment information in Pay for Order flow with ECE/PRB

### ISSUE THREAD

**Issue by cesarcosta99** [Automattic/woocommerce-payments#9048]
### Description

In the Pay for Order flow, the order data is not being populated with payment information when using Express Checkout Element (ECE). This results in incomplete order records, affecting the merchant's ability to track orders accurately.

This issue is also present when using Payment Request Buttons (PRBs).

### Acceptance criteria

- Order data is populated with information provided during the ECE payment.
- If PRBs haven't been removed yet, we should extend this enhancement to them.

### Testing instructions

> [!IMPORTANT]
> You must test this flow on a public facing URL using HTTPS. Use something like Jurassic Tube or ngrok to create a tunnel to your local environment.

1. Enable ECE by setting the `_wcpay_feature_stripe_ece` option to 1.
2. As a merchant, navigate to **Payments > Settings**.
3. Enable Apple Pay/Google Pay and save changes.
4. Navigate to **WooCommerce > Orders > Add order**.
5. Click on **Add item(s)**, then on **Add product(s)**.
6. Type in some product name, such as "Beanie" and select it.
7. Click on the **Add** button, and finally, on the **Create** button.
8. Copy the **Customer payment page** link.
9. As a shopper, navigate to the link you copied.
10. Pay using ECE and make sure the payment is processed successfully.
11. As a merchant, navigate to the order you created in step 7.
12. Note in the order notes that the payment has been successfully charged, but the  **Billing** and **Shipping** sections are empty.

**Comment by frosso** [comment]
The shipping information is not asked for (and hence not provided) in the pay-for-order flow when using the PRB/ECE: https://github.com/Automattic/woocommerce-payments/blob/145da0a528a08beebb1376312cc7a09e37466ed1/client/express-checkout/index.js#L562
The Billing and Shipping address information should be filled in by the merchant when creating the order.

If we were to overwrite the billing information in the order with the billing information provided by the ECE, we'd be creating a discrepancy in behavior with all the other payment methods.
In the case of cards or other Stripe Payment Element payment methods for example, we're not overwriting the billing information.
In the case of other extensions like PayPal w/ its own wallet, we're also not overwriting the billing information from the payment method itself.

If we were to implement such a change, I think it should be provided as an option for merchants to opt in.

**Comment by pierorocca** [comment]
I agree with @frosso's interpretation. @cesarcosta99 great observation that we could turn into an enhancement. Could you think about what that might look like? e.g. change the pay for order flow so that merchant either must pre-fill the info or collect it from the shopper. e.g. auto-populate if blank but otherwise do not overwrite existing information.

**Comment by cesarcosta99** [comment]
@pierorocca @frosso I understand the interpretation here. But I think we should consider approach shipping and billing information differently in the pay for order flow, and maybe regular flows when paying with ECE/PayPal/other wallets too.

>The Billing and Shipping address information should be filled in by the merchant when creating the order.

When a merchant is creating an order, they may have shipping information and can fill them out. However, the customer can pay for the order in many ways, and the billing information, which always depend on the payment method used, may or may not differ from the shipping info. Therefore, even though merchants can (and have that info, in some cases), it's not expected from them to provide billing information to the order.

>If we were to overwrite the billing information in the order with the billing information provided by the ECE, we'd be creating a discrepancy in behavior with all the other payment methods.
In the case of cards or other Stripe Payment Element payment methods for example, we're not overwriting the billing information.
In the case of other extensions like PayPal w/ its own wallet, we're also not overwriting the billing information from the payment method itself.

In the regular checkout flow (i.e. handled by Stripe Payment Element), customers are able to provide shipping and billing info. In the ECE, and I believe other wallets flows too, checkout doesn't require any information (typed) from the user and the billing information is attached to the payment method. Both are different cases when it comes to order information and maybe it's time to revisit how we handle these. IMO, if customer has not provided billing info during checkout, it should be ok to retrieve it from the payment method and overwrite the order billing info, unless there is some data handling concern I'm not aware of.

**Comment by pierorocca** [comment]
I was thinking about this from a card perspective. I need to rethink about ECE.

For card: I was thinking it as sort of like the WooPay NUX flow. If the merchant fills in all the info, it's all prepopulated and everything is collapsed. If the merchant adds only shipping, the shopper must enter the card and billing info. If the merchant only adds billing, the shopper must enter the shipping. If the merchant enters nothing and just sends a fill it in yourself type of order, then the shopper fills in everything but the email. I don't think this can work because that allows a shopper to change what the merchant entered which takes control out of their hands and could open up fraud. e.g. change shipping address to my house while I use a stolen credit card with the address the merchant entered.

I created an order with no email, no shipping, no billing and I'm confused about this use case. I am buying a hat that can't be shipped to me. Zip code is an acceptable alternative to a full billing address. The impact is that less information means a lower approval rate or it could be blocked by the advanced fraud rules if that requires a full address for the AVS check.

Looking at a Pay by Link use case which I created on Stripe, it's possible to sell a product and not collect a shipping or billing address. To @frosso point, they allow the merchant to opt in to collecting shipping, billing, or both. They do however force collection of email.

Donations could be another use case where shipping nor billing info is needed.

This is response is getting long, I think the order page needs to support options to collect addresses if the merchant isn't providing them or doesn't have them. Thoughts?

**Comment by cesarcosta99** [comment]
Yeah, there's a wide variety of use cases and donations was one I hadn't thought about. After seeing your use case listing, having an option to collect the shipping address makes more sense to me now, but maybe this is better suited to WC core?

As for the billing address, is there any problem in pulling it from the payment method to add it to the order? When I think about donations, I see that not being necessary, but for regular stores may be helpful.
